src/mlir/generator.cc

```diff
@@ -397,25 +417,47 @@ namespace mlir::verona
       return std::move(err);
 
     // Store the value in the alloca
-    auto op =
-      genOperation(getLocation(ast), "verona.store", {*rhs, store}, unkTy);
+    auto op = genOperation(
+      getLocation(ast), "verona.store", {rhs->get(), store}, unkTy);
     if (auto err = op.takeError())
       return std::move(err);
     return store;
   }
 
-  llvm::Expected<mlir::Value> Generator::parseCall(const ::ast::Ast& ast)
+  llvm::Expected<ReturnValue> Generator::parseCall(const ::ast::Ast& ast)
   {
     assert(isCall(ast) && "Bad node");
     auto name = getID(ast);
 
     // All operations are calls, only calls to previously defined functions
-    // are function calls. FIXME: Is this really what we want?
-    if (functionTable.inScope(name))
+    // are function calls.
+    if (auto func = functionTable.lookup(name))
     {
-      // TODO: Lower calls.
-      return parsingError(
-        "Function calls not implemented yet", getLocation(ast));
+      auto argNodes = getAllOperands(ast);
```

Comment:
With the change above, this could then be:
```suggestion
      llvm::SmallVector<::ast::WeakAst, 8> argNodes;
      getAllOperands(argNodes, ast);
```

And now we can choose to use any vector-like type.

Reply:
Because of the template parameter, this can only work if the implementation is in the header, not the `cc` file.

Given this is only a prototype for the interface, I don't think we should worry about it that much.